tests/ci/run_s2n_interop_test.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+set -ex
+
+source tests/ci/common_posix_setup.sh
+
+scratch_folder=${SYS_ROOT}/"s2n-scratch"
+s2n_url='https://github.com/aws/s2n-tls.git'
+s2n_branch='main'
+lc_url='https://github.com/aws/aws-lc.git'
+lc_branch='main'
+
+mkdir -p "${scratch_folder}"
+rm -rf "${scratch_folder:?}"/*
+
+# clone s2n-tls
+git clone --depth 1 --branch "${s2n_branch}" "${s2n_url}" "${scratch_folder}/s2n-tls"
+
+# clone aws-lc
+git clone --depth 1 --branch "${lc_branch}" "${lc_url}" "${scratch_folder}/s2n-tls/aws-lc"
+
+# build aws-lc
+echo "building aws-lc"
+cd "${scratch_folder}/s2n-tls/aws-lc"
+cmake -GNinja -B build
+ninja -C build
+cmake --install build --prefix install
+
+# build s2n-tls with aws-lc
+echo "building s2n_tls"
+cd "${scratch_folder}/s2n-tls"
+cmake . -Bbuild-with-lc \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DCMAKE_PREFIX_PATH=aws-lc/install
+cmake --build build-with-lc
+
+# handshake test 1 - aws-lc bssl server with s2n-tls s2nc client for X25519MLKEM768:X25519
```

Comment:
Add a comment explaining how all this works. It's not immediately apparent what all these workarounds do. e.g. the `sleep` is there to allow the server time to listen on the socket.